src/pages/iv-items/forms/SaleList.js

```diff
@@ -0,0 +1,269 @@
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { useForm } from 'src/hooks/form.js'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { Grid } from '@mui/material'
+import SalesForm from './SalesForm'
+import FormShell from 'src/components/Shared/FormShell'
+
+const SalesList = ({ store, labels, maxAccess, formikInitial }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { recordId } = store
+
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  const { formik } = useForm({
+    initialValues: {
+      currencyId: '',
+      defSaleMUId: store.measurementId || '',
+      pgId: store.priceGroupId || '',
+      returnPolicyId: store.returnPolicy || ''
+    },
+
+    enableReinitialize: true,
+    validateOnChange: true,
+    onSubmit: async obj => {
+      const submissionData = {
+        ...formikInitial,
+        defSaleMUId: formik.values.defSaleMUId,
+        returnPolicyId: formik.values.returnPolicyId
+      }
+
+      await postRequest({
+        extension: InventoryRepository.Items.set,
+        record: JSON.stringify(submissionData)
+      })
+    }
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      if (recordId) {
+        try {
+          const response = await getRequest({
+            extension: InventoryRepository.Currency.qry,
+            parameters: `&_itemId=${recordId}`
+          })
+          if (response.list && response.list.length > 0) {
+            formik.setFieldValue('currencyId', response.list[0].currencyId)
+          }
+        } catch (error) {}
+      }
+    })()
+  }, [recordId])
+
+  async function fetchGridData() {
+    if (formik.values.currencyId) {
+      return await getRequest({
+        extension: SaleRepository.Sales.qry,
+        parameters: `&_itemId=${recordId}&_currencyId=${formik.values.currencyId}`
+      })
+    }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    invalidate,
+    refetch
+  } = useResourceQuery({
+    enabled: !!recordId,
+    datasetId: ResourceIds.SalesList,
+    queryFn: fetchGridData,
+    endpointId: SaleRepository.Sales.qry
+  })
+
+  useEffect(() => {
+    if (formik.values.currencyId) {
+      ;(async () => {
+        await fetchGridData(formik.values.currencyId)
```

Comment:
remove `if(formik.values.currencyId)`
and here pass `fetchGridData(formik?.values?.currencyId)`